conductor_api/src/holochain.rs

```diff
@@ -108,14 +107,15 @@ use holochain_core_types::{
 
 use holochain_json_api::json::JsonString;
 
+use holochain_core::state::StateWrapper;
 use jsonrpc_core::IoHandler;
 use std::sync::Arc;
 
 /// contains a Holochain application instance
 pub struct Holochain {
-    instance: Instance,
+    instance: Option<Instance>,
     #[allow(dead_code)]
-    context: Arc<Context>,
+    context: Option<Arc<Context>>,
```

Comment:
I would like to reinstate the `ugly::Initable` type for these, and other places where a `None` corresponds to an invalid state rather than a truly optional value that was omitted. Then we can be reminded of that fact, and hopefully eventually refactor this so that we can't represent these invalid states anymore, and can avoid them altogether. In the meantime, I figure we will still run into panics every now and then, it's very hard to catch all the edge cases.

Reply:
Well, these options don't represent invalid states. They enable an explicit drop of the content:
```rust
    pub fn kill(&mut self) {
        let _ = self.stop();
        self.instance = None;
        self.context = None;
    }
```